### ISSUE TITLE
Brave's loading of pages on startup are slowed down more and more by filter rules added to the adblocker.

### ISSUE THREAD

**Issue by Emi-Emi-Emi** [brave/brave-browser#36031]
## Description 

@antonok-edm 

I noticed this because I modified the new `list_catalog.json` and `regional_catalog.json`, to include the Default lists in `brave://settings/shields/filters` like any regional list and being able to easy disable them.

I did this in order to easily add default lists as custom lists, and be able to trouble shoot adblocker better.

The problem is, when the browser opens, the pages will start loading after seconds... too many seconds, it doesn't matter if they are heavy websites like Youtube or just Brave Search, it will take long time to start loading.

This has to do with the amount of adblocker filter rules, because while I know there should be a difference by loading thousands of rules, and the updates and all, it gets bad the more custom rules with more rules you add.
There is even a huge difference between Brave loading a page on startup with default adblocking settings on vs the adblocker disabled, and when you add custom filters with thousands of rules, well, it seems to get too slow not to notice this.

It has to do with the number of rules being loaded for sure but also the 'engine' where they are being added. because just using Easylist will make the startup being slower.

Also, this doesn't seem to affect the normal loading of pages after the startup delay, so might just be a startup issue, but I didn't test it deeply.
I am sure with something like this https://github.com/brave/brave-browser/issues/36007 this type of test will be easier to test and see how much negative impact the adblocker has in the browser, at least, this one is pretty easy to tell 👍

## Steps to Reproduce

   1. Load `https://easylist.to/easylist/easylist.txt` and `https://easylist.to/easylist/easyprivacy.txt` and any big list as custom list, the more the longer it will take.
   2. make sure to enable `Continue where you left off` in brave://settings/getStarted
   3. go to https://brave.com/
   4. now `brave://restart`
   5. Now when Browser restarts, it will take too many seconds to load brave page.
   6. Disable the custom lists, restart and now it should take a 'normal time' to load the page.
   7. Disable `Trackers & ads blocking` in brave://settings/shields and restart and everything loads immediately with no delay at all.

## Brave version (brave://version info)

1.64.70 Chromium: 122.0.6261.29 (Official Build) nightly (64-bit)
Windows 11 2H22 22621.3155

**Comment by antonok-edm** [comment]
May be related to https://github.com/brave/brave-browser/issues/35922; I'm looking into a fix.

**Comment by Emi-Emi-Emi** [comment]
Yeah, seems like it should be that.
Although my CPU stays at 13% the whole time on my side, so it is not high high, but it only goes down when the pages start loading.

Since I disabled default lists, only regional lists should update on my side, yet, Custom lists in brave://settings/shields/filters, which should depend on Expires property or update only every 7 days, start freaking out on start, like if they were being enabled and disabled and updated, something I reported here https://github.com/brave/brave-browser/issues/28894 and then the page is unusable until it finishes collapsing and expanding doing whatever it is doing with the custom lists.

This is how it looks and how long it takes on my side to start loading pages, so you can see what I mean, although I am sure you are full aware of it all.

https://cdn.gilcdn.com/ContentMediaGenericFiles/2cac6c281a0372b11fca22851ca24ee2-Full.mp4

**Comment by Uneccessary** [comment]
I hope this issue will get fixed in the near future...

**Comment by Emi-Emi-Emi** [comment]
Closing this, as it doesn't seem to be as much issue now as back then. Brave still takes some seconds to start loading pages after a restart, but not as before.

We have to remember this original issue was happening because I was disabling Default lists and then I noticed running the same list as Custom lists for troubleshooting reasons was causing seconds for any page to load, since they are 140K+ rules.
Another reason for me to run Default lists as custom was because running lists in the same 'engine' will get the benefit of running exceptions rules like  `#@#.CSSselector`, `#@#+js()`, `$badfilter`, so I was taking advantage of that as well.

Since then, I found a better approach of using custom rules where I can add exceptions to Default lists, but without using Default lists as custom, which means it won't sacrifice the startup time and performance, also I get the benefits like Default lists automatically excluding `!#if` that aren't compatible with Chromium/Brave, that was recently pushed to the component updater.

So I can't see anyone having issues with this unless they load thousands of rules in the custom lists, and that can only happen if people run Default lists as custom as I did voluntarily, or thinking Brave doesn't Easylist or uBlock lists so they add them thinking they are getting better blocking with them as custom; or they found other third party adblock lists to add. but I don't think they would be near the many thousand, and if they do, I doubt they don't have a bunch of duplicates Default lists and regional list already cover.

Thank you.
